### PR TITLE
add readInlineData

### PR DESCRIPTION
**Pull request by itajaja**
tested locally. should I add tests?

### CODE REVIEW

**Comment by itajaja** [comment]
cc @taion @jquense

**Comment by taion** [comment]
- should update version to v6, since this is in relay v6
- does the relay compiler distinguish inline fragments from normal fragments at all? possibly not, i'm sure, but it'd be neat if it did

**Comment by itajaja** [comment]
> should update version to v6, since this is in relay v6

the version is not specified, except here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38430/files#diff-9bf0e4126d019d96bea44f758a214a1aR1

**Review by alloy — CHANGES_REQUESTED** [review]
Awesome ✨ 

Yes, some test coverage would be great, especially around the fragment reference checking.

File: types/react-relay/test/react-relay-tests.tsx [hunk 1]
```diff
@@ -5,17 +5,19 @@ import * as React from 'react';
 import { Environment, Network, RecordSource, Store, ConnectionHandler } from 'relay-runtime';
 
 import {
-    graphql,
     commitMutation,
     createFragmentContainer,
     createPaginationContainer,
     createRefetchContainer,
-    requestSubscription,
+    graphql,
```
Comment (itajaja) [inline types/react-relay/test/react-relay-tests.tsx]: I just sorted the imports alphabetically

**Comment by itajaja** [comment]
the only question I have is that by using `readInlineFragment`, the consumer who is writing the function will need `_FragmentRefs`, which, given the _ sounds a little internal (even though is exported). I can't think of a better alternative to type that function, so I think that it might be better to rename that type?

File: types/react-relay/test/react-relay-tests.tsx (hunk 1 above)
Comment (alloy) [reply types/react-relay/test/react-relay-tests.tsx]: That's fine, prettier was added to DT recently-ish 👍

File: types/react-relay/test/react-relay-tests.tsx [hunk 2]
```diff
@@ -594,6 +596,16 @@ function markNotificationAsRead(source: string, storyID: string) {
...
 
+// ~~~~~~~~~~~~~~~~~~~~~
+// readInlineData
+// ~~~~~~~~~~~~~~~~~~~~~
+
+function functionWithInline(storyRef: _FragmentRefs<Story_story>): Story_story {
+    return readInlineData(graphql``, storyRef)
+}
+
+functionWithInline({ ' $fragmentRefs': 1 as any });
```
Comment (alloy) [inline types/react-relay/test/react-relay-tests.tsx]: Can you please use a more realistic example and no casting to `any`?

**Comment by alloy** [comment]
Yeah I think you’re right and the consumer will need to use that type, so re-naming sounds good 👍

**Comment by itajaja** [comment]
> Yeah I think you’re right and the consumer will need to use that type, so re-naming sounds good

I acutally changed to use `FragmentOrRegularProp` so no need to rename

File: types/react-relay/test/react-relay-tests.tsx [hunk 3]
```diff
@@ -594,6 +596,18 @@ function markNotificationAsRead(source: string, storyID: string) {
...
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~
+// readInlineData
+// ~~~~~~~~~~~~~~~~~~~~~
+
+function functionWithInline(
+  storyRef: FragmentOrRegularProp<Story_story>,
+): Story_story {
+  return readInlineData<Story_story>(graphql``, storyRef);
```
Comment (alloy) [inline types/react-relay/test/react-relay-tests.tsx]: In this example, would the `Story_story` fragment be defined in this (empty) `graphql` template string?

(And, if so, can you add it? 🙏)

Comment (itajaja) [reply types/react-relay/test/react-relay-tests.tsx]: sure, but I don't think that would in any way affect the tests. as `graphql` returns a tagged node whatever the string content is. I just saw other usages of `graphql`` ` so I just used. I can change it for the sake of clarity/documentation though

Comment (alloy) [reply types/react-relay/test/react-relay-tests.tsx]: My request is about the latter.

**Review by alloy — APPROVED** [review]

**Comment by alloy** [comment]
Perfect, thanks for bearing with me! 🙏

**Pull request merged**